Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts

```diff
@@ -176,21 +179,41 @@ export class KeyVaultClient extends azureServiceClient.ServiceClient {
                 '{secretName}': secretName
             }
         );
-
         console.log(tl.loc("DownloadingSecretValue", secretName));
-        this.invokeRequest(httpRequest).then(async (response: webClient.WebResponse) => {
-            if (response.statusCode == 200) {
-                var result = response.body.value;
-                return new azureServiceClientBase.ApiResult(null, result);
-            }
-            else if (response.statusCode == 400) {
-                return new azureServiceClientBase.ApiResult(tl.loc('GetSecretFailedBecauseOfInvalidCharacters', secretName));
-            }
-            else {
-                return new azureServiceClientBase.ApiResult(azureServiceClientBase.ToError(response));
-            }
-        }).then((apiResult: azureServiceClientBase.ApiResult) => callback(apiResult.error, apiResult.result),
-            (error) => callback(error));
+        this.invokeRequestWithRetry(httpRequest,secretName)
+    .then((apiResult: azureServiceClientBase.ApiResult) => callback(apiResult.error, apiResult.result))
+    .catch(error => callback(error));
+
+    }
+
+    async invokeRequestWithRetry(httpRequest, secretName, retrysCount: number = 3, retryWait: number = 2000) {
+        return this.invokeRequest(httpRequest).then(async (response: webClient.WebResponse) => {
+             // Retry logic
+        let attempts = 0;
+                if (retrysCount > 0) {
+                    try {
+                        // Check response status code
+                        if (response.statusCode === 200) {
+                            var result = response.body.value;
+                            return new azureServiceClientBase.ApiResult(null, result);
+                        } else if (response.statusCode === 400) {
+                            return new azureServiceClientBase.ApiResult(tl.loc('GetSecretFailedBecauseOfInvalidCharacters', secretName));
+                        } else {
+                            return new azureServiceClientBase.ApiResult(azureServiceClientBase.ToError(response));
+                        }
+                    } catch (error) {
+                        console.error(tl.loc(`Attempt ${attempts + 1} failed with error: ${error}`));
+                        attempts++;
+                        console.log(tl.loc(`Retrying ...`));
+                            await new Promise(r => setTimeout(r, retryWait));
+                            return await this.invokeRequestWithRetry(httpRequest,secretName, retrysCount-1, retryWait)
+                        
+                    }
+                }
+                throw new Error(`Failed after ${attempts} attempts`);
```

Comment:
attempts here will be always 1. Use retrysCount to track attempt number

Reply:
modified the logic to 4-Retriescount